llvm/lib/CodeGen/ExpandMemCmp.cpp

```diff
@@ -656,6 +658,40 @@ Value *MemCmpExpansion::getMemCmpOneBlock() {
 
   const LoadPair Loads = getLoadPair(LoadSizeType, BSwapSizeType, MaxLoadType,
                                      /*Offset*/ 0);
+
+  // If a user of memcmp cares only about two outcomes, for example:
+  //    bool result = memcmp(a, b, NBYTES) > 0;
+  // We can generate more optimal code with a smaller number of operations
+  if (auto *U = CI->getUniqueUndroppableUser()) {
```

Comment:
You are looking for hasOneUse(). If you use getUniqueUndroppableUser() you actually need to drop droppable users.